ide/db/src/org/netbeans/modules/db/explorer/DbSchemasQuery.java

```
@@ -0,0 +1,73 @@
+
+import java.sql.DatabaseMetaData;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import org.netbeans.api.db.explorer.DatabaseException;
+import org.openide.util.Exceptions;
+
+/**
+ * 
+ * @author Jan Horvath
+ */
+public class DbSchemasQuery {
+
+    /**
+     * Retrieves a list of database schemas available for given connection.
+     * 
+     * @param dbconn
+     * @return List of schemas or null if connection failed
+     */
+    public static List<String> getSchemasForConnection(org.netbeans.api.db.explorer.DatabaseConnection dbconn) {
+        List<String> schemas = null;
+        try {
+            DatabaseConnection databaseConnection = new DatabaseConnection();
+            databaseConnection.setDriver(dbconn.getDriverClass());
+            databaseConnection.setDatabase(dbconn.getDatabaseURL());
+            databaseConnection.setUser(dbconn.getUser());
+            databaseConnection.setPassword(dbconn.getPassword());
+            databaseConnection.setSchema(dbconn.getSchema());
+            databaseConnection.setConnectionProperties(dbconn.getConnectionProperties());
+            
+            databaseConnection.connectSync();
+            if (databaseConnection.getState() == DatabaseConnection.State.connected) { 
+                DatabaseMetaData dbMetaData = databaseConnection.getJDBCConnection().getMetaData();
+                if (dbMetaData.supportsSchemasInTableDefinitions()) {
+                    ResultSet rs = dbMetaData.getSchemas();
+                    if (rs != null) {
+                        while (rs.next()) {
+                            if (schemas == null) {
+                                schemas = new ArrayList<>();
+                            }
+                            schemas.add(rs.getString(1).trim());
+                        }
+                    }
+                }
+            }
+        } catch (DatabaseException | SQLException ex) {
```

Why not propagate the exception to the caller?